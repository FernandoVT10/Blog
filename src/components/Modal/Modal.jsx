import { useEffect } from "react";

import "./Modal.scss";

export default ({ children, title, active = false, onClose, prefix = "" }) => {
    useEffect(() => {
        $(`#${prefix}-modal`).on("hidden.bs.modal", () => {
            onClose();
        });
    }, []);
    
    useEffect(() => {
        if(active) {
            $(`#${prefix}-modal`).modal("show");
        } else {
            $(`#${prefix}-modal`).modal("hide");
        }
    }, [active]);
    
    return (
        <div className="modal fade" tabIndex="-1" role="dialog" id={`${prefix}-modal`}>
            <div className="modal-dialog modal-xl" role="document">
                <div className="modal-content modal__content">
                    <div className="modal-header modal__header">
                        <h2 className="modal__title">{ title }</h2>

                        <button
                        type="button"
                        className="close modal__close-button"
                        data-dismiss="modal"
                        aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                        </button>
                    </div>
                    <div className="modal-body modal__body">
                        { children }
                    </div>
                </div>
            </div>
        </div>
    );
};