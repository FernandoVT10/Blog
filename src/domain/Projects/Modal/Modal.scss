.modal {
    &__content {
        background-color: var(--black);
        color: var(--white);
    }

    &__header {
        border: none;
    }

    &__title {
        font-size: 1.2em;
        font-weight: bold;
    }

    &__close-button {
        outline: 0 !important;
        color: var(--secondary);
        transition: .3s;

        &:hover {
            color: var(--secondary);
        }
    }

    &__footer {
        border: none;
    }
}
