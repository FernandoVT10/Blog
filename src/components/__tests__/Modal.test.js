import Modal from "../Modal/";
import { create } from "react-test-renderer";

window.$ = () => ({ on: jest.fn(), modal: jest.fn() });

test("Check if <Modal/> renders correctly", () => {
    const component = create(
        <Modal title="Test modal title" modalSize="modal-xl" prefix="modalTest">
            <p>Test children</p>
        </Modal>
    );

    expect(component.toJSON()).toMatchSnapshot();
});