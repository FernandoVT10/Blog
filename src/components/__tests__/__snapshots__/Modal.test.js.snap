// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <Modal/> renders correctly 1`] = `
<div
  className="modal fade"
  id="modalTest-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-xl"
    role="document"
  >
    <div
      className="modal-content modal__content"
    >
      <div
        className="modal-header modal__header"
      >
        <h5
          className="modal__title"
        >
          Test modal title
        </h5>
        <button
          aria-label="Close"
          className="close modal__button"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body modal__body"
      >
        <p>
          Test children
        </p>
      </div>
    </div>
  </div>
</div>
`;
